src/odemis/driver/scanner.py

```diff
@@ -112,3 +124,211 @@ def updateMetadata(self, md):
 
     def getMetadata(self):
         return self._external.getMetadata()
+
+    def claimBeam(self, claim):
+        """
+        Used to claim the beam by blanking the beam on the scanner of the XT client when the blanker
+        mode is set to automatic (blanker.value = None)
+        :param claim (boolean): True for unblanking and False for blanking
+        """
+        if self.blanker.value is None:
+            self._setBlanker(not claim)  # Set the blanker using the setter of the VA
+
+
+class CompositedDetector(model.Detector):
+    '''
+    A generic Detector which takes 2 dependencies to create a one detector. It's
+    essentially a wrapper to an Detector to generate data using the "external"
+    detector while manipulating the "internal" detector.
+    '''
+    def __init__(self, name, role, dependencies, **kwargs):
+        '''
+        dependencies (dict string->model.HwComponent): the dependencies
+            There must be exactly two dependencies "external" and "internal".
+        Raise:
+          ValueError: if the dependencies are not compatible
+        '''
+        # we will fill the set of dependencies with Components later in ._dependencies
```

Comment:
This comment doesn't match what you're doing. The next line actually creates a .dependencies VA.